Sources/MongoSwift/MongoCursor.swift

```diff
@@ -197,37 +192,65 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
      *   - `DecodingError` if an error occurs decoding the server's response.
      */
     public func nextOrError() throws -> T? {
-        if let next = self.next() {
-            return next
-        }
-        if let error = self.error {
-            throw error
+        let next = self.next()
+        if let next = next {
+            switch next {
+            case let .success(doc):
+                return doc
+            case let .failure(error):
+                throw error
+            }
         }
         return nil
     }
 
-    /// Returns the next `Document` in this cursor, or `nil`. After this function returns `nil`, the caller should use
-    /// the `.error` property to check for errors. For tailable cursors, users should also check `isAlive` after this
-    /// method returns `nil`, to determine if the cursor has the potential to return any more data in the future.
-    public func next() -> T? {
+    /**
+     * Returns an array of type  `Document` from this cursor, or throws
+     * an error if any of the `Result` objects contain an error.
+     * - Returns: an array of type `Document`
+     * - Throws:
+     *   - `ServerError.commandError` if an error occurs on the server while iterating the cursor.
+     *   - `UserError.logicError` if this function is called after the cursor has died.
+     *   - `UserError.logicError` if this function is called and the session associated with this cursor is inactive.
+     *   - `DecodingError` if an error occurs decoding the server's response.
+     */
+    public func all() throws -> [T] {
+        return try self.map {
+            switch $0 {
+            case let .success(t):
+                return t
+            case let .failure(error):
+                throw error
+            }
+        }
+    }
+
+    /// Returns a `Result` containing the next `Document` in this cursor, `nil`
```

Comment:
1) similar to my comment above, this can be any `Codable` type `T`

2) I think it would be good if this docstring also listed the possible errors that can occur calling `next`, as you have above for `nextOrError`.